tests/unit/test_analytics.py

```diff
@@ -1,6 +1,40 @@
+import json
 import pytest
+import mock
 
-from dvc.analytics import Analytics
+from dvc import analytics
+from dvc.utils.compat import builtin_str, str
+
+
+@pytest.fixture
+def tmp_global_config(tmp_path):
+    with mock.patch(
+        "dvc.config.Config.get_global_config_dir", return_value=tmp_path
+    ):
+        yield
+
+
+def test_collect(tmp_global_config):
```

Comment:
Why do we need `tmp_global_config` here? If that is to not pollute actual global config then let's say that in a comment. Or in a `tmp_global_config` docstring.

Reply:
Because `collect` calls, `find_or_create_user_id()`.

I'll add a comment)